### PR TITLE
Progress bar UX Improvements

### PR DESCRIPTION
**Pull request by Aryex82**
Issue: #436

Description: 
1. Created a new component for progress bar
2. Currently under feature flag.
3. Also added two extra dependencies ember-modifiers and lodash

video: 

https://github.com/Real-Dev-Squad/website-my/assets/98796547/ab800804-aeae-4fd6-a675-d3e70d9ee301

Test Coverage:
Screenshot 2023-07-21 at 10 13 55 PM

### CODE REVIEW

File: app/components/progress-bar.hbs
Comment (yesyash) [inline app/components/progress-bar.hbs:1]: - if we're following BEM convention should we change this to `progress-bar__container` ?

File: app/components/progress-bar.js
Comment (yesyash) [inline app/components/progress-bar.js:4]: - Any reason for adding such a `lodash` apart from using the debounce function it provides? 
Can we write this by ourself ?

File: app/components/progress-bar.js
Comment (yesyash) [inline app/components/progress-bar.js:18]: - Instead of having all the logic inside the if statement, moving the date delta into its own variable would look much nicer. 
```js
timeDelta = Date.now() - this.lastEditedTime
if (this.isEditable && timeDelta >= 5000){
    ...
}
```
what do you think?

File: app/modifiers/focus-when.js
Comment (yesyash) [inline app/modifiers/focus-when.js:3]: - should we change the name to something else from `foucsWhen` as it's not very descriptive, what about `setFocus` as this sets the focus on a element.

Comment (yesyash) [inline app/modifiers/focus-when.js:3]: - should we change the name to something else from `foucsWhen` as it's not very descriptive, what about `setFocus` as this sets the focus on a element.

File: app/styles/progress-bar.css
Comment (yesyash) [inline app/styles/progress-bar.css:15]: - please use definite values instead of percentages for font size, as percentage can have very different values across devices and screen resolutions.

**Review by yesyash — CHANGES_REQUESTED** [review]

File: app/components/progress-bar.hbs
Comment (Aryex82) [reply app/components/progress-bar.hbs]: made required changes.

File: app/components/progress-bar.js
Comment (Aryex82) [reply app/components/progress-bar.js]: Removed loads and added a debounce method.
Just curious any other reason for this other than bundle size increase?

File: app/components/progress-bar.js
Comment (Aryex82) [reply app/components/progress-bar.js]: made required changes.

File: app/modifiers/focus-when.js
Comment (Aryex82) [reply app/modifiers/focus-when.js:3]: I feel it does make the hbs file more readable:

File: app/styles/progress-bar.css
Comment (Aryex82) [reply app/styles/progress-bar.css]: made required changes.

File: app/components/progress-bar.hbs
Comment (prakashchoudhary07) [inline app/components/progress-bar.hbs:47]: Please move svg to assets and use it from there

File: app/components/progress-bar.js
Comment (yesyash) [reply app/components/progress-bar.js]: - I personally prefer using a library if it solves your problem which you otherwise have taken you much longer to solve, apart from the bundle size adding a library also requires you to maintain it over the times as new there are new updated made to it. 

Having multiple libraries also increases your build time as there is more code for the bundler to go through.

File: app/modifiers/focus-when.js
Comment (prakashchoudhary07) [reply app/modifiers/focus-when.js:3]: I also think setFocus would be a better naming choice

File: app/components/progress-bar.hbs
Comment (Aryex82) [reply app/components/progress-bar.hbs]: I am dynamically changing the colour of the svg here. In this case would it be ok? Or is there any other way to do the same? 

Screenshot 2023-07-25 at 11 59 31 PM

File: app/components/progress-bar.js
Comment (Aryex82) [reply app/components/progress-bar.js]: done.

File: app/utils/debounce.js
Comment (Aryex82) [reply app/utils/debounce.js]: done

**Review by Pratiyushkumar — COMMENTED** [review]
you should also handle it after the user has stopped updating the progress on the progress bar then the progress bar should get disabled

File: tests/integration/components/progress-bar-test.js
Comment (Pratiyushkumar) [inline tests/integration/components/progress-bar-test.js:1]: please give the test case name's a meaningful ones
for example - 'it has edit button' --> this does not tell me anything meaningful, I am not getting what are you exactly testing

File: app/components/progress-bar.js
Comment (Aryex82) [reply app/components/progress-bar.js]: created an issue to use later in future as tests were failing: https://github.com/Real-Dev-Squad/website-my/issues/460

File: tests/integration/components/progress-bar-test.js
Comment (Aryex82) [reply tests/integration/components/progress-bar-test.js:1]: done.

**Review by prakashchoudhary07 — APPROVED** [review]

File: tests/integration/components/progress-bar-test.js
Comment (Aryex82) [reply tests/integration/components/progress-bar-test.js:1]: I changed the other test case names. not sure what to rename this test case to 'it has edit button'. because it only checks whether the progress bar has an edit button or not.

**Comment by Aryex82** [comment]
> you should also handle it after the user has stopped updating the progress on the progress bar then the progress bar should get disabled

it has that functionality.

File: tests/integration/components/progress-bar-test.js
Comment (Aryex82) [reply tests/integration/components/progress-bar-test.js:1]: Sounds good. I am working on an refactor issue #460. Will make the changes there.

Comment (Pratiyushkumar) [reply tests/integration/components/progress-bar-test.js:1]: image

no changes visible to me yet....please make sure changes are done before asking for review.

Comment (Aryex82) [reply tests/integration/components/progress-bar-test.js:1]: Yes since it's a nitpick I don't want to dismiss an approval. So Will make these changes on a PR. Ive mentioned the issue ticket above.

File: app/components/progress-bar.js
Comment (yesyash) [inline app/components/progress-bar.js:26]: - why are we adding this delay ?

Comment (Aryex82) [reply app/components/progress-bar.js:26]: When we click the edit button the progress bar becomes editable and edit button shows progress percent. And if user doesn't perform any actions on the progress bar then this timer reverts everything to original state.

**Review by yesyash — APPROVED** [review]

**Pull request merged**
